src/BuildLoop.cpp

```diff
@@ -150,27 +160,234 @@ static void SignalMainThreadToStartCleaningUp(BuildQueue *queue)
+    remove(digestString);
+
+    uint64_t now = TimerGet();
+    double duration = TimerDiffSeconds(time_exec_started, now);
+
+    MutexLock(&queue->m_Lock);
+    PrintMessage(writeResult == CacheResult::Success ? MessageStatusLevel::Success : MessageStatusLevel::Warning, duration, "%s [CacheWrite %s]", node->m_DagNode->m_Annotation.Get(), digestString);
+    MutexUnlock(&queue->m_Lock);
+}
+
+static NodeBuildResult::Enum ExecuteNode(BuildQueue* queue, RuntimeNode* node, Mutex *queue_lock, ThreadState* thread_state, StatCache* stat_cache, const Frozen::DagDerived* dagDerived)
+{
+    CheckDoesNotHaveLock(&queue->m_Lock);
+
+    if (IsNodeCacheableByLeafInputsAndCachingEnabled(queue,node))
+    {
+        CHECK(node->m_CurrentLeafInputSignature != nullptr);
+        bool stillTheSame = node->m_BuiltNode && node->m_BuiltNode->m_WasBuiltSuccessfully && node->m_BuiltNode->m_LeafInputSignature == node->m_CurrentLeafInputSignature->digest;
+
+        if (!stillTheSame)
+            if (!VerifyAllVersionedFilesIncludedByGeneratedHeaderFilesWereAlreadyPartOfTheLeafInputs(queue, thread_state, node, dagDerived))
+                return NodeBuildResult::kRanFailed;
+    }
+
+    bool haveToRunAction = CheckInputSignatureToSeeNodeNeedsExecuting(queue, thread_state, node);
+    if (!haveToRunAction)
+        return NodeBuildResult::kUpToDate;
+
+    NodeBuildResult::Enum runActionResult = RunAction(queue, thread_state, node, queue_lock);
+
+    if (runActionResult == NodeBuildResult::kRanSuccesfully && queue->m_Config.m_AttemptCacheWrites && IsNodeCacheableByLeafInputsAndCachingEnabled(queue,node))
+        AttemptCacheWrite(queue,thread_state,node);
+
+    return runActionResult;
+}
+
+static bool AttemptToMakeConsistentWithoutNeedingDependenciesBuilt(RuntimeNode* node, BuildQueue* queue, ThreadState* thread_state)
+{
+    CheckHasLock(&queue->m_Lock);
+
+    if (node->m_BuiltNode)
+    {
+        if (node->m_BuiltNode->m_LeafInputSignature == node->m_CurrentLeafInputSignature->digest && !OutputFilesMissingFor(node->m_BuiltNode, queue->m_Config.m_StatCache) && node->m_BuiltNode->m_WasBuiltSuccessfully)
+        {
+            node->m_BuildResult = NodeBuildResult::kUpToDate;
+            FinishNode(queue, node);
+            return true;
+        }
+    }
+
+    RuntimeNodeSetAttemptedCacheLookup(node);
+
+    uint64_t time_exec_started = TimerGet();
+    MutexUnlock(&queue->m_Lock);
+    auto cacheReadResult = CacheClient::AttemptRead(queue->m_Config.m_Dag, node->m_DagNode, node->m_CurrentLeafInputSignature->digest, queue->m_Config.m_StatCache, &queue->m_Lock, thread_state);
+    MutexLock(&queue->m_Lock);
+
+    uint64_t now = TimerGet();
+    double duration = TimerDiffSeconds(time_exec_started, now);
+    char digestString[kDigestStringSize];
+    DigestToString(digestString, node->m_CurrentLeafInputSignature->digest);
+
+    switch (cacheReadResult)
+    {
+        case CacheResult::DidNotTry:
+            break;
+
+        case CacheResult::Failure:
```

Comment:
Should we also make this the default case for the switch statement?

Reply:
Fixed